VocaDbWeb/Views/SongList/Details.cshtml

```diff
@@ -88,12 +88,20 @@
 						<a href="@Url.EntryDetails(ev, ev.UrlSlug)">
 							@ev.Name
 						</a>
-						@if (ev.Date.HasValue && !string.IsNullOrEmpty(ev.VenueName)) {
-							<small>(@ev.Date.Value.ToString("d"), @ev.VenueName)</small>
+						@if (ev.Date.HasValue) {
+							if (ev.Venue != null) {
+								<small>(@ev.Date.Value.ToString("d"), <a href="@Url.EntryDetails(ev.Venue)">@ev.Venue.Name</a>)</small>

```

Comment:
I'm thinking you could make choosing venue entry vs. venue name into some sort of helper (maybe in EventHelpers.cshtml), because it's used in many places.

Reply:
Yeah, I was thinking just the same thing. I thought it would be better to do that after replacing SongListDetailsContract with SongListForApiContract.

Reply:
Perhaps it would be better to replace SongListDetailsContract with SongListForApiContract after moving all entry images under the static domain (#16).

Reply:
You could also just add interfaces, similar to IUser. It might be a good idea anyway.

Reply:
Can I add a new property for URL to original picture to EntryThumbForApiContract? If not, how can I show the original picture on the song list details page? Perhaps we should not replace SongListDetailsContract with SongListForApiContract for now. (Additionally, we'll need some new interfaces and classes like ISongList and IUserExtender. By the way, how about renaming `-Extender` classes to `-Extensions` after merging #583?)

Reply:
I'll try to [refactor image URL handling a bit](https://github.com/VocaDB/vocadb/commit/6e043d735b540ab7e68ca258de413b17d7604167), so we could add that original image URL to EntryThumbForApiContract.

Reply:
Huh, I refactored the image URL generation, which ended up being a bit more than expected (which is common). Anyway, EntryThumbForApiContract now has original image URL, and the URL generation is much more straightforward in general. Just use the "aggregate" image URL factory.

Reply:
Thank you! I'll try to update this PR in the next few days.